torch/package/package_exporter.py

```diff
@@ -929,10 +1001,25 @@ def _execute_dependency_graph(self):
                 if attrs.get("is_pickle") is True:
                     # This node came from save_source_pickle, we don't need to write any source for it.
                     continue
-
                 is_package = attrs["is_package"]
                 source = attrs["source"]
                 self._write_source_string(module_name, source, is_package)
+            elif action == _ModuleProviderAction.SELECTIVE_EXTERN:
+                # pdb.set_trace()
+                is_package = attrs["is_package"]
+                source = attrs["source"]
+                # pdb.set_trace()
+                if is_package and source:
+                    selective_extern_template_file = str(Path(__file__).parent / "_selective_extern.py")
+                    selective_extern_file_template = Template(_read_file(selective_extern_template_file))
+
+                    matches = self._selective_interns[module_name]
+                    original_init = self._get_source_of_module(self._import_module(module_name))
```

Comment:
We should not be saving a copy of the original `__init__` to the package. The contract for this feature is that, for a given selectively externed module (e.g. `torch`), it behaves exactly as if it were externed normally *except* if you choose to access a selectively interned module (e.g. `torch.fb`).

For an example of how things would break: we reference lots of symbols defined in our C extension in `torch/__init__.py`, like `_C._set_deterministic_algorithms`. If we ever renamed those, old packages would break (since the original init code that we preserved would reference the old name). 

For another example: module initialization often manipulates global state (like we do in `torch/__init__.py` by setting the determinism of our numerical algorithms), so running the init code twice (once internally from the package, and once in the externed version of package) can have unpredictable results.

The previous PR for selective intern had some implementation where we didn't copy the init iirc.

Reply:
I actually lifted this from the original PR actually. This makes a lot of sense, I was also a bit confused of why it was there.